### PR TITLE
feat(savings): convex finance

### LINKED ISSUE #5779: [Bounty] Integrate Convex, A Yield Farming Platform into Mask Plugin
## Catalogue

- Platform
  - [x] Gitcoin
  - [ ] Hackerlink
  - [ ] Mask
- Experience
  - [ ] Beginner
  - [x] Intermediate
  - [ ] Advanced
- Category
  - [ ] Mask Core Feature
  - [x] Mask Plugin
  - [ ] Security/Cryptography Feature
  - [ ] Documentation
  - [ ] Design
  - [ ] Other
    - [ ] /_List it here_/
- Time Commitment
  - [x] 2 weeks
- Bounty Size
  - [x] 200 $MASK or 1000 USDC
## Part 1: Problem Description
Background: Convex Finance, a platform built to boost rewards for CRV stakers and liquidity providers alike, all in a simple and easy to use interface. Convex aims to simplify staking on Curve, as well as the CRV-locking system with the help of its native fee-earning token: CVX.

If you’ve ever been a Curve LP, you know it is somewhat non-trivial to maximize your boost by depositing/maintaining your veCRV balance. If you’ve never been a Curve LP, it may be intimidating to do so without being a DeFi power user. Convex aims to make this process easy and bring the CRV boost ecosystem to everyone.

Liquidity providers earn trading fees and claim boosted CRV without locking CRV themselves. Convex has no deposit or withdrawal fees, and a low performance fee, which is distributed to CRV stakers and CVX token holders.

Solution: Integrate Convex under the savings button in the entry page.
The savings page will list all savings options provided by AAVE, Compound, YFI ,etc and include both stablecoin savings and non-stable token savings (staking/farming etc).

See our mock design in the link, does not need to be 1 to 1 but should follow the flow
https://www.figma.com/file/gVkQ67y285b4FXVV1KPThN/TwitterV1?node-id=17600%3A374185

_The Entry Page_
image

_The Savings Page_

image

_The Supply Page_

image

_Convex Convert and Stake Page_

Screen Shot 2022-03-04 at 12 44 08 AM

Please condense the Convex info into the existing design. Please talk to us about ideas regarding design.

Required Verbiage to help spread the word: "I just deposited xxx dollars into xxx. Will I make money this week?"

## Part 2: Resources

- [Bounty Guide](https://github.com/DimensionDev/Maskbook/tree/master/docs/bounty-guide.md)
- https://docs.convexfinance.com/convexfinance/
- https://github.com/convex-eth/platform

## Part 3: Contact and Support
@yisiliu @tzhan28 @LindaMask for ideas
@Jack-Works for implementations
Mask Network discord server: https://discord.gg/4Xccj7a75Q,
Convex discord server: https://discord.com/invite/uAwvZfs9qU

Please speak up in the Mask discord's bounty section or just discord dm @yisiliu or @taylor zhang or @lin_dao #5779  so that we can find you on Discord. You can also reach out to lin_DAO on telegram and she will create a dedicated group for faster communication

Submission Requirements
Please include demo to show the integration is working.

Judging Criteria
Have a working demo.
Clean code & tech design
Clear UI, easy to use and generally good user experience

### PR DESCRIPTION
**Pull request by developerfred**
## Description

close #5779

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by guanbinrui** [comment]
Hi @developerfred Any updates?

**Comment by developerfred** [comment]
> Hi @developerfred Any updates?

I will update the plugins saving today

File: cspell.json [hunk 1]
```diff
@@ -15,19 +15,27 @@
     "words": [
         "aave",
         "adai",
+        "addresss",
```
Comment (Jack-Works) [inline cspell.json]: seriously? 🤔

Comment (developerfred) [reply cspell.json]: 🤦🏾‍♂️ sorry i will remove it

**Comment by guanbinrui** [comment]
Hi @developerfred Plz resolve those conflicts.

File: packages/mask/src/plugins/Savings/protocols/ConvexProtocol.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        const poolID = poolData?.id!
+        const poolInfo = await BoostContract?.methods.poolInfo(poolID).call()
+        const tokenPooladdress = poolInfo?.token
+
+        const stakingToken = createContract<ICurveFi>(web3, tokenPooladdress || ZERO_ADDRESS, CurveFiABI as AbiItem[])
+        const stakingTokenERC20 = createContract<ERC20>(web3, tokenPooladdress || ZERO_ADDRESS, ERC20ABI as AbiItem[])
+
+        const allowance = await stakingTokenERC20?.methods.allowance(account, boostAddress).call()
+
+        // // check allowance
```
Comment (septs) [inline packages/mask/src/plugins/Savings/protocols/ConvexProtocol.ts]: plz remove unused code

Comment (developerfred) [reply packages/mask/src/plugins/Savings/protocols/ConvexProtocol.ts]: done

**Comment by Jack-Works** [comment]
`packages/mask/src/plugins/Savings Old/SNSAdaptor/assets/convex.svg`?

**Comment by developerfred** [comment]
> `packages/mask/src/plugins/Savings Old/SNSAdaptor/assets/convex.svg`?

I will point to the correct folder path

**Comment by guanbinrui** [comment]
Failed to build.

**Comment by AinuX** [comment]
@developerfred 
CI build all failed, please fix, thanks!

**Comment by Jack-Works** [comment]
ERROR in ./src/plugins/Savings/SNSAdaptor/IconURL.tsx 5:27-74
Module not found: Error: Can't resolve './assets/convex.png' in '/home/circleci/repo/packages/mask/src/plugins/Savings/SNSAdaptor'

**Pull request closed without merging**